compiler/AST/foralls.cpp

```diff
@@ -810,6 +828,26 @@ static void checkForNonIterator(IteratorGroup* igroup, bool gotSA,
   }
 }
 
+// If iterFn is a serial iterator, make it look parallel, as downstream
+// passes depend on that. I.e. label with FLAG_INLINE_ITERATOR.
+// Create a clone to avoid potential impact on non-parallel uses.
```

Comment:
Is this creating a parallel version or something? Looks really work-around-y. If it's a workaround, maybe just say it's a workaround in the comment?